File: backend/Models/BranchInventory.cs
using System;

namespace backend.Models
{
    public class BranchInventory
    {
        public int BranchId { get; set; }
        public int ProductId { get; set; }
        public int QuantityOnHand { get; set; }
        public int ReorderLevel { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }
    }
}

File: api-backend/app/Models/BranchInventory.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class BranchInventory extends Model
{
    

    protected $table = 'branch_inventory';
    protected $primaryKey = ['branch_id', 'product_id'];
    public $incrementing = false;
    public $timestamps = true;

    protected $fillable = [
        'branch_id',
        'product_id',
        'quantity_on_hand',
        'reorder_level',
        'last_updated_by',
    ];

    protected $casts = [
        'quantity_on_hand' => 'integer',
        'reorder_level' => 'integer',
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
        'deleted_at' => 'datetime',
    ];

    public function branch()
    {
        return $this->belongsTo(Branch::class, 'branch_id');
    }

    public function product()
    {
        return $this->belongsTo(Product::class, 'product_id');
    }

    public function updater()
    {
        return $this->belongsTo(Employee::class, 'last_updated_by');
    }
}